cirq/_compat.py

```diff
@@ -180,3 +181,34 @@ def decorated_func(*args, **kwargs) -> Any:
         return decorated_func
 
     return decorator
+
+
+def wrap_module(module: ModuleType,
+                deprecated_attributes: Dict[str, Tuple[str, str]]):
+    """Wrap a module with deprecated attributes.
+
+    Args:
+        module: The module to wrap.
+        deprecated_attributes: A dictionary from attribute name to pair of
+            strings, where the first string gives the version that the attribute
+            will be removed in, and the second string describes what the user
+            should do instead of accessing this deprecated attribute.
+
+    Returns:
+        Wrapped module with deprecated attributes.
+    """
+
+    class Wrapped(ModuleType):
+
+        def __getattr__(self, name):
+            if name in deprecated_attributes:
+                deadline, fix = deprecated_attributes[name]
```

Comment:
Nit: You can unpack tuples in one line. `if name, (deadline, fix) in deprecated_attributes.items():`
Not important to fix for this PR.